src/pages/nft/claim/index.vue

```diff
@@ -729,13 +729,19 @@ export default {
     },
     loginAddress(newValue) {
       this.claimingAddress = newValue;
+      if (this.claimingAddress && this.state === NFT_CLAIM_STATE.LOGIN) {
+        this.state = NFT_CLAIM_STATE.ID_CONFIRMATION;
+      }
       if (!this.getAddress && !(this.status === 'completed')) {
         this.claimingAddress = '';
       }
     },
     state(newValue) {
-      if (newValue) {
-        this.$router.push({ query: { ...this.$route.query, state: newValue } });
+      if (newValue === NFT_CLAIM_STATE.LOGIN && this.claimingAddress) {
+        this.state = NFT_CLAIM_STATE.ID_CONFIRMATION;
+        this.navigateToState(NFT_CLAIM_STATE.ID_CONFIRMATION);
+      } else if (newValue) {
```

Comment:
It seems that this logic could be done in `navigateToState` instead of a watcher, if it's generic to any state change

Reply:
[🎨 Utilize navigateToState for all state management](https://github.com/likecoin/liker-land/pull/1616/commits/0e26cf2d3f56535f180a83b571367a852b85e796)